src/main/scala/com/kinja/amqp/AmqpClientFactory.scala

```diff
@@ -1,26 +1,40 @@
 package com.kinja.amqp
 
 import com.kinja.amqp.persistence.{ InMemoryMessageBufferDecorator, MessageStore }
-
-import akka.actor.{ ActorSystem, ActorRef }
+import akka.actor.{ ActorRef, ActorSystem }
 import com.github.sstone.amqp.ConnectionOwner
 import com.rabbitmq.client.ConnectionFactory
-import java.sql.Connection
+import utils.Utils._
+
 import org.slf4j.Logger
-import scala.concurrent.ExecutionContext
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.control.NonFatal
 
 class AmqpClientFactory {
+
+	def createConsumerClient(
+		config: AmqpConfiguration,
+		actorSystem: ActorSystem,
+		logger: Logger,
+		ec: ExecutionContext,
+		connectionListener: Option[ActorRef]
+	): AmqpConsumerClientInterface =
+		createClient(config, actorSystem, logger, ec, Map.empty[AtLeastOnceGroup, MessageStore], connectionListener)
+
 	def createClient(
 		config: AmqpConfiguration,
 		actorSystem: ActorSystem,
 		logger: Logger,
 		ec: ExecutionContext,
-		messageStores: Map[AtLeastOnceGroup, MessageStore]
+		messageStores: Map[AtLeastOnceGroup, MessageStore],
```

Comment:
I'm wondering if we should make absolutely impossible to create a producer client without any message stores. We could do that by changing this constructor to expect `messageStore: (AtLeastOnceGroup, MessageStore)` and then we could set up additional message stores using this pattern:

```scala
createClient(..., firstMessageStore, ...)
  .addMessageStore(secondMessageStore)
  .addMessageStore(thirdMessageStore)
```

Or an even simpler solution: make this a variable argument similarly to how the `NonEmptyList` constructor works:

```scala
def createClient(
   ...
   messageStore: (AtLeastOnceGroup, MessageStore),
   additionalMessageStores: (AtLeastOnceGroup, MessageStore)*
)
```

Reply:
I like your second API proposal more. If nobody protest against it I'm changing the API.

Reply:
@balagez What is your opinion about the following:

```
	def createClient(
		config: AmqpConfiguration,
		actorSystem: ActorSystem,
		logger: Logger,
		ec: ExecutionContext,
		messageStore: (AtLeastOnceGroup, MessageStore),
		connectionListener: Option[ActorRef]
	): AmqpClientInterface =
		createClient(config,actorSystem,logger,ec, Map(messageStore), connectionListener)

	def createClient(
		config: AmqpConfiguration,
		actorSystem: ActorSystem,
		logger: Logger,
		ec: ExecutionContext,
		messageStores: ::[(AtLeastOnceGroup, MessageStore)],
		connectionListener: Option[ActorRef]
		): AmqpClientInterface =
		createClient(config,actorSystem,logger,ec, messageStores.toMap, connectionListener)

	private def createClient(
		config: AmqpConfiguration,
		actorSystem: ActorSystem,
		logger: Logger,
		ec: ExecutionContext,
		messageStores: Map[AtLeastOnceGroup, MessageStore],
		connectionListener: Option[ActorRef]
	)
```

Reply:
TBH I have never seen this `::[T]` notation before in a public API. It might be a clever solution but just looking at the function definition you need think twice what's happening. It's not expressing the type you expect there because what you expect is basically a non-empty list. But you did not write that, what you wrote is an implementation detail of `List`. Also how do users create an instance of it? They're forced to use the same notation because that's the only way you can create an instance, so you're adding clutter to client code too. I don't think that `::` is meant to be used this way, I believe there is a reason Scalaz and Cats added their `NonEmptyList` types instead of just pointing put that the standard Scala library already has it. So I can't say I truly like it.

@migmit-kinja What do you think?

Reply:
I think that `::[]` (not an emoji), while cool, lacks certain features — like, for example, there are convenient `List` and `NonEmptyList` constructors, but there isn't a better and readily available way to create, say, `1 :: 2 :: 3 :: Nil` — `List.apply` returns `List`, not `::` (which is a shame — they could've easily added two variants of `List.apply`, with 0 arguments and with at least one).

Also, it's not very well known (which is sad), and would cause a bit of confusion.

Reply:
What are your opinion @balagez @migmit-kinja about using our NonEmptyList library in the next version? (I have an other metrics sending issue what I want to fix anyway)

Reply:
I think it's OK

Reply:
`NonEmptyList` is in a private repo, this one is public so it won't work. Also I'm a huge enemy of library dependencies because if you do that, and change `NonEmptyList`, you'll need to release not only that, but then this library, and then EVERY SINGLE APPS we have. So please don't use the `NonEmptyList` library, simply use one of the solutions I recommended. They don't add any code and maintenance cost.